scripts/setup-ssh-keys.sh

```
@@ -0,0 +1,767 @@
+    export ACTIVE_SSH_KEY
+    export GREEN RED NC
+    
+    if [ "$HAS_PARALLEL" = true ]; then
+        # Use GNU parallel for distribution
+        printf '%s\n' "${DISCOVERED_HOSTS[@]}" | parallel -j "$PARALLEL_JOBS" distribute_to_host
+    else
+        # Use background jobs for parallel execution
+        local pids=()
+        for host_entry in "${DISCOVERED_HOSTS[@]}"; do
+            # Limit concurrent jobs
+            while [ ${#pids[@]} -ge $PARALLEL_JOBS ]; do
```

Shellcheck (0.10.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)